binlog_streamer.go

```diff
@@ -212,10 +221,11 @@ func (s *BinlogStreamer) Run() {
 			// As a result, the following case will set the last resumable position for
 			// interruption to EITHER the start (if using GTIDs) or the end of the
 			// last transaction
-			s.lastResumableBinlogPosition.Pos = uint32(ev.Header.LogPos)
-			s.updateLastStreamedPosAndTime(ev)
-		default:
```

Comment:
since we're removing the `default` case, are we are certain that we won't be missing events that will get us to the `stopAtBinlogPosition`?

Reply:
the default case is still the same.. all events are "valid" position by default (through the `isEventPositionValid` variable)

Reply:
I have the same question. It may be possible that `stopAtBinlogPosition` points to one of the events where `isEventPositionValid == false`, thereby causing a hang, likely in test only.

Reply:
There is now only two explicit cases where `isEventPositionValid == false`. They're clearly documented with references to the mysql source. All other events are valid positions.